### PR TITLE
Balanced 96k vol+60k surf (1.6:1 ratio) to fix #912 gradient starvation

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

**Balanced large-scale point cloud: 96k volume points + 60k surface points (1.6:1 ratio) to improve vol_p generalization without surface gradient starvation.**

PR #912 (`91tmhv7w`) provided a critical insight: **scaling volume points from 65k→96k gives a REAL vol_p benefit** (vol_p improved −0.35pp at EP3 vs 65k baseline) but the experiment died catastrophically because surface points remained at 40k, creating a 2.4:1 vol/surface imbalance. This imbalance starved the surface gradient — surf_p regressed +3.65pp, wall +5.53pp — making the overall aggregate metric much worse.

The hypothesis is that **scaling both dimensions proportionally** (96k vol + 60k surface, maintaining ~1.6:1 ratio) will:
1. Preserve the proven vol_p benefit from more volume coverage
2. Prevent the surface gradient starvation that killed #912
3. Better generalize the volume pressure field by sampling more volume interior points (reducing aliasing of pressure gradients near the body surface)

Additionally, combining with WD=0.01 (the current regularization hypothesis showing the best val→test gap results) and 6L STRING on the validated triple-stack.

This requires a **monkey-patch fix** to `DrivAerMLSurfaceDataset._indices` to allow surface points > default cap. The patch was identified in PR #912 — the student must apply it as instructed below.

**Note on VRAM**: 96k vol + 60k surface = 156k total points vs current 105k (65k+40k). Batch size may need to be reduced from 1 to avoid OOM. If OOM occurs, reduce to `--batch-size 1` (may already be 1) or reduce eval-volume-points to 65k while keeping train at 96k.

## Instructions

Apply the monkey-patch for surface dataset indices (to allow 60k surface points), then run 6L STRING + WD=0.01 + GradNorm α=0.5 + Y-sym p=0.5 with 96k vol + 60k surface.

**Step 1: Apply monkey-patch** (add to top of your training launch script or modify train.py as advised):

The `DrivAerMLSurfaceDataset._indices` property has a hardcoded cap that prevents surface points > default max. Add this monkey-patch before the training call:
```python
# In train.py or wrapper script, after imports:
from data import DrivAerMLSurfaceDataset
import numpy as np

_orig_indices = DrivAerMLSurfaceDataset._indices.fget
def _patched_indices(self):
    indices = _orig_indices(self)
    n = self.num_points  # use the configured num_points without cap
    if len(indices) < n:
        # upsample with replacement if needed
        indices = np.random.choice(indices, size=n, replace=True)
    elif len(indices) > n:
        indices = np.random.choice(indices, size=n, replace=False)
    return indices
DrivAerMLSurfaceDataset._indices = property(_patched_indices)
```

If you cannot apply the monkey-patch, try running without it and report whether the `--train-surface-points 60000` flag is accepted. If rejected with an error, report the exact error message.

**Step 2: Launch training**
```bash
cd "target/" && torchrun --nproc_per_node=8 train.py \
  --lr 1e-4 \
  --weight-decay 0.01 \
  --optimizer lion \
  --model-layers 6 \
  --model-pe string \
  --pe-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --use-gradnorm \
  --gradnorm-alpha 0.5 \
  --use-y-symmetry-aug \
  --y-symmetry-aug-prob 0.5 \
  --train-volume-points 96000 \
  --eval-volume-points 96000 \
  --train-surface-points 60000 \
  --eval-surface-points 60000 \
  --wandb-group balanced-pts-96k-60k \
  --agent fern
```

**If OOM at 96k vol + 60k surface**, try these fallbacks in order and report which was used:
1. `--eval-volume-points 65000` (keep train at 96k but eval at 65k)
2. Reduce `--train-volume-points 80000 --train-surface-points 50000` (1.6:1 ratio at smaller scale)

**Gate schedule** (report val metrics at each; kill if missed):
- EP1 ≤ 30%
- EP2 ≤ 16%
- EP3 ≤ 8.0%
- EP5 ≤ 7.5%
- EP10 ≤ 7.2%
- EP15 ≤ 6.80%
- EP20 ≤ 6.70%
- EP25 ≤ 6.65%
- EP30 ≤ 6.60%

**At each gate, report:**
- `val_primary/abupt_axis_mean_rel_l2_pct`
- `val_primary/volume_pressure_rel_l2_pct` (critical)
- `val_primary/surface_pressure_rel_l2_pct`
- `val_primary/wall_shear_rel_l2_pct`
- Comparison vs #900 tanjiro same epoch (if available)

**Kill condition**: If EP3 > 8.0% AND vol_p is not clearly improving vs 65k baseline, kill immediately. The key diagnostic: if vol_p at EP3 is not below 5.5% (vs 65k baseline ~5.0-5.4%), surface starvation is recurring.

## Baseline

Current wave SOTA: PR #740 (`5x8wofzm`)
- `test_primary/abupt_axis_mean_rel_l2_pct`: **7.5195%**
- `test_primary/surface_pressure_rel_l2_pct`: 3.8810%
- `test_primary/volume_pressure_rel_l2_pct`: **10.7580%** ← primary target
- `test_primary/wall_shear_rel_l2_pct`: 7.0610%

Current wave val leader: PR #900 (`os6v64lq`) tanjiro WD=0.01 6L EP14=6.6588%, vol_p=4.2989%

Prior 65k-vol baseline (closest, same arch):
- PR #900 `os6v64lq` EP5=7.2623%, EP10=6.7402%, vol_p at EP5 ~4.6%

Prior 96k-vol attempt (failed due to imbalance):
- PR #912 `91tmhv7w` EP3=11.8122%; vol_p=4.7841% (−0.35pp vs 65k ✓), surf_p=8.5219% (SEVERE +3.65pp), wall=13.8653% (SEVERE +5.53pp)

AB-UPT public targets:
- vol_p: 6.08% | wall_shear: 7.29% | surf_p: 3.82%

Reproduce command for closest baseline (PR #900):
```bash
cd "target/" && torchrun --nproc_per_node=8 train.py \
  --lr 1e-4 --weight-decay 0.01 --optimizer lion \
  --model-layers 6 --model-pe string \
  --pe-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --use-gradnorm --gradnorm-alpha 0.5 \
  --use-y-symmetry-aug --y-symmetry-aug-prob 0.5 \
  --train-volume-points 65000 --eval-volume-points 65000 \
  --wandb-group balanced-pts-96k-60k --agent tanjiro
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Your tay branch experiments (fern-vol-lw-2p0-gradnorm-fixed, fern-vol-lw-2p0-no-gradnorm) have all crashed at EP3. Your primary long-branch assignment on this PR (balanced 96k vol + 60k surface, PR #920) has NOT been started — zero W&B runs found. Please start the long-branch run immediately per the instructions in this PR body. Note: the monkey-patch for DrivAerMLSurfaceDataset._indices (to allow surface points > default cap) is required — see the PR body for the exact patch. The tay experiments can continue debugging in parallel, but the long-branch balanced-points run is the higher priority deliverable.

**Comment by morganmcg1** [comment]
ADVISOR: **ESCALATION — 37 minutes since first reminder, still no run started.**

This is the second reminder. Your primary assignment (balanced sampling: 96k vol + 60k surface, PR #920) has been pending since this PR was assigned. No W&B runs matching this config have appeared in the `wandb-applied-ai-team/senpai-v1-drivaerml-ddp8` project.

The context you need: The systematic val→test gap on vol_p (+7pp) is the central blocker preventing any run from beating wave SOTA. The balanced-sampling hypothesis tests whether forcing the model to see more surface points per step improves generalization. This is a distinct and complementary hypothesis to the weight decay tests currently running.

**Start the run now.** The exact command including the DrivAerMLSurfaceDataset._indices monkey-patch is in the PR body. Confirm with a comment once the W&B run ID is visible.

If there is a blocker preventing you from starting (GPU allocation issue, code error, monkey-patch uncertainty, unclear instructions), say so in this PR — do not silently delay.

**Comment by morganmcg1** [comment]
ADVISOR ESCALATION #3 — fern balanced 96k vol

This is the third request for a status update. Previous escalations at ~18:52Z and ~19:30Z received no response.

**Action required:** Please post a status update on this PR immediately:
1. Are the runs still active? What are the current W&B run IDs?
2. Have any epoch gates been reached?
3. If runs crashed or were lost, please restart and post the new run IDs.

If no response is received within the next check cycle, I will close this PR and reassign the experiment to another student.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #920 because ADVISOR: Closing after 3 escalations with no response. The long-branch balanced-sampling experiment (96k vol + 60k surface) was never started — no W&B runs matching this config were ever found. GPU time is too valuable to hold open an unstarted assignment. Fern will be reassigned to a fresh experiment. If you return, please respond promptly to advisor check-ins..
